component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java

```diff
@@ -0,0 +1,235 @@
+                        description = "Channel of the MQ server which should use to connect",
+                        type = DataType.STRING),
+                @Parameter(name = IBMMQConstants.QUEUE_MANAGER_NAME,
+                        description = "Name of the Queue Manager",
+                        type = DataType.STRING),
+                @Parameter(name = IBMMQConstants.USER_NAME,
+                        description = "User name of the server",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "will try to connect without username"),
+                @Parameter(name = IBMMQConstants.PASSWORD,
+                        description = "Password of the server",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "will try to connect without username"
+                ),
+        },
+        examples = {
+                @Example(description = "This example shows how to connect to an IBM MQ queue and "
+                        + "receive messages.",
+                        syntax = "@source(type='ibmmq',"
+                                + "destination.name='Queue1',"
+                                + "host='192.168.56.3',"
+                                + "port='1414',"
+                                + "channel='Channel1',"
+                                + "queue.manager = 'ESBQManager',"
+                                + "password='1920',"
+                                + "username='mqm',"
+                                + "@map(type='text'))"
+                                + "define stream SweetProductionStream(name string, amount double);"),
+        }
+)
+
+public class IBMMQSink extends Sink {
+    private static final Logger log = LoggerFactory.getLogger(IBMMQSink.class);
+    private OptionHolder optionHolder;
+    private QueueConnection connection;
+    private MQQueueConnectionFactory connectionFactory;
+    private StreamDefinition outputStreamDefinition;
+    private QueueSession session;
+    private Queue queue;
+    private QueueSender messageSender;
+    private MessageConsumer consumer;
+    private String userName;
+    private String password;
+    private boolean isSecured = false;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class, Map.class, ByteBuffer.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    @Override
+    protected void init(StreamDefinition outputStreamDefinition, OptionHolder optionHolder, ConfigReader
+            sinkConfigReader, SiddhiAppContext siddhiAppContext) {
+        this.optionHolder = optionHolder;
+        connectionFactory = new MQQueueConnectionFactory();
+        this.userName = optionHolder.validateAndGetStaticValue(IBMMQConstants.USER_NAME, "null");
```

Comment:
Can't we assign null and later do a simple null check instead of comparing strings?